src/astroid.cc

```diff
@@ -158,6 +163,16 @@ namespace Astroid {
     if (vm.count ("disable-log")) {
       LOG (warn) << "disabling log";
       logging::core::get()->set_logging_enabled (false);
+    } else if (vm.count ("log-level")) {
+      ustring llevel = vm["log-level"].as<ustring> ();
+
+      /* Not very elegant map for string to enum */
+      std::map<std::string, logging::trivial::severity_level> sevmap = boost::assign::map_list_of
+        ("trace", logging::trivial::trace)("debug", logging::trivial::debug)
+        ("info", logging::trivial::info)("warning", logging::trivial::warning)
+        ("error", logging::trivial::error)("fatal", logging::trivial::fatal);
+
+       logging::core::get()->set_filter (logging::trivial::severity >= sevmap[llevel]);
```

Comment:
What if llevel is not one of the legal options?

Reply:
it will be silently ignored

Reply:
ok, maybe put in a comment